src/TraceEvent/TraceLog.cs

```diff
@@ -153,10 +153,199 @@ public static TraceLog OpenOrConvert(string etlOrEtlxFilePath, TraceLogOptions o
+                session.Stop();
+                task.Wait();
+            }
+        }
+
+        /// <summary>
+        /// Attaches callbacks to initialize modules and methods based on the rundown events.
+        /// Unlike SetupCallbacks(), these callbacks are attached before any events are processed so
+        /// they do not rely on ForAllUnresolvedCodeAddressesInRange() and instead record all the methods.
+        /// </summary>
+        private void SetupInitialRundownCallbacks(EventPipeEventSource rawEvents)
+        {
+            rawEvents.HeadersDeserialized += delegate () { CopyHeadersFrom(rawEvents); };
+
+            var clrRundownParser = new ClrRundownTraceEventParser(rawEvents);
+            Action<ModuleLoadUnloadTraceData> onLoaderRundown = delegate (ModuleLoadUnloadTraceData data)
+            {
+                TraceProcess process = processes.GetOrCreateProcess(data.ProcessID, data.TimeStampQPC);
+                process.LoadedModules.ManagedModuleLoadOrUnload(data, false, true);
+            };
+
+            clrRundownParser.LoaderModuleDCStop += onLoaderRundown;
+            clrRundownParser.LoaderModuleDCStart += onLoaderRundown;
+
+            // TODO We don't seem to be getting any of these events, only MethodDCStopVerbose.
+            // clrRundownParser.MethodDCStartVerbose += delegate (MethodLoadUnloadVerboseTraceData data)
+            // {
+            //     if (data.IsJitted)
+            //     {
+            //         TraceProcess process = processes.GetOrCreateProcess(data.ProcessID, data.TimeStampQPC);
+            //         process.InsertJITTEDMethod(data.MethodStartAddress, data.MethodSize, delegate ()
+            //         {
+            //             TraceManagedModule module = process.LoadedModules.GetOrCreateManagedModule(data.ModuleID, data.TimeStampQPC);
+            //             MethodIndex methodIndex = CodeAddresses.Methods.NewMethod(TraceLog.GetFullName(data), module.ModuleFile.ModuleFileIndex, data.MethodToken);
+            //             return new TraceProcess.MethodLookupInfo(data.MethodStartAddress, data.MethodSize, methodIndex);
+            //         });
+
+            //         jittedMethods.Add((MethodLoadUnloadVerboseTraceData)data.Clone());
+            //     }
+            // };
+
+            clrRundownParser.MethodILToNativeMapDCStop += delegate (MethodILToNativeMapTraceData data)
+            {
+                codeAddresses.AddILMapping(data);
+            };
+
+            clrRundownParser.MethodDCStopVerbose += delegate (MethodLoadUnloadVerboseTraceData data)
+            {
+                // TODO we need this also for non-jitted methods, otherwise we won't resolve some frames, for example:
+                //      "System.Private.CoreLib.il" - "System.Threading.Tasks.Task.Wait()"
+                //      Is it OK to use InsertJITTEDMethod & FindJITTEDMethodFromAddress or do we need something else?
+                // if (data.IsJitted)
```

Comment:
It should be fine to use the JITTED method calls for all MethodDCStopVerbose events.

Reply:
OK, changed the TODO to a comment and cleaned up  there.